processout-sdk/src/main/java/com/processout/processout_sdk/ProcessOut.java

```diff
@@ -82,59 +97,200 @@ public void updateCvc(Card card, final CvcUpdateCallback callback) {
+     *
+     * @param invoiceId Invoice ID that should be used for charging (this needs to be generated on your backend).
+     *                  Keep in mind that you should set the return_url to "your_app://processout.return".
+     *                  Check https://www.docs.processsout.com for more details
+     * @param callback  Callback for listing alternative payment methods
      */
     public void listAlternativeMethods(final String invoiceId, final ListAlternativeMethodsCallback callback) {
         final Gson gson = new Gson();
         final Context context = this.context;
         final String projectId = this.projectId;
 
 
-        Network.getInstance(this.context, this.projectId).CallProcessOut("/gateway-configurations?filter=alternative-payment-methods&expand_merchant_accounts=true", Request.Method.GET, null, new Network.NetworkResult() {
+        Network.getInstance(this.context, this.projectId).CallProcessOut(
+                "/gateway-configurations?filter=alternative-payment-methods&expand_merchant_accounts=true",
+                Request.Method.GET, null, new Network.NetworkResult() {
+                    @Override
+                    public void onError(Exception error) {
+                        callback.onError(error);
+                    }
+
+                    @Override
+                    public void onSuccess(JSONObject json) {
+                        try {
+                            JSONArray configs = json.getJSONArray("gateway_configurations");
+
+                            ArrayList<AlternativeGateway> gways = new ArrayList<>();
+                            for (int i = 0; i < configs.length(); i++) {
+                                AlternativeGateway g = gson.fromJson(
+                                        configs.getJSONObject(i).toString(), AlternativeGateway.class);
+                                g.setContext(context);
+                                g.setProjectId(projectId);
+                                g.setInvoiceId(invoiceId);
+                                gways.add(g);
+                            }
+
+                            callback.onSuccess(gways);
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                            callback.onError(e);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * Allow card payments authorization (with 3DS2 support)
+     *
+     * @param invoiceId previously generated invoice
+     * @param source    source to use for the charge (card token, etc.)
+     * @param handler   (Custom 3DS2 handler)
+     */
+    public void makeCardPayment(final String invoiceId, final String source, final ThreeDSHandler handler) {
+        try {
+            // Generate the authorization body and forces 3DS2
+            final JSONObject body = new JSONObject(
+                    "{\"source\":\"" + source + "\", \"enable_three_d_s_2\": true}");
```

Comment:
I'm not a fan of building JSON payloads like this— why not having a proper structure for it?